Source/Csla.Generators/cs/Csla.Generators.CSharp/AutoSerializable/IncrementalSerializationPartialsGenerator.cs

```diff
@@ -0,0 +1,50 @@
+using System.Text;
+using Csla.Generators.CSharp.AutoSerialization.Discovery;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Csla.Generators.CSharp.AutoSerialization
+{
+  [Generator]
+  public class IncrementalSerializationPartialsGenerator : IIncrementalGenerator
+  {
+    public void Initialize(IncrementalGeneratorInitializationContext context)
+    {
+      // Register a syntax receiver to collect information during the initial parsing phase
+      var classDeclarations = context.SyntaxProvider.ForAttributeWithMetadataName(
+              "Csla.Serialization.AutoSerializableAttribute",
+
+              predicate: static (s, _) => (s is ClassDeclarationSyntax || s is StructDeclarationSyntax) && s is TypeDeclarationSyntax type && type.Modifiers.Any(SyntaxKind.PartialKeyword),
+              transform: static (ctx, _) =>
+              {
+                if (ctx.TargetNode is TypeDeclarationSyntax typeDeclarationSyntax)
+                {
+                  return TypeDefinitionExtractor.ExtractTypeDefinition(new DefinitionExtractionContext(ctx.SemanticModel), typeDeclarationSyntax);
+                }
+                return null;
+              })
+          .Where(static m => m is not null);
+
+      // Combine the collected syntax nodes with the semantic model
+      var compilationAndClasses = context.CompilationProvider.Combine(classDeclarations.Collect());
```

Comment:
This is highly inefficient because _every_ keypress forces the source generator to rerun even though it's probably not affected.
I can recommend this and the whole series around that post: https://andrewlock.net/creating-a-source-generator-part-9-avoiding-performance-pitfalls-in-incremental-generators/#5-be-careful-using-compilationprovider

Reply:
Thanks 
I changed it to only select the nullable from the compilation.

While @TheCakeMonster  validate the generated source I'll take a look in the others pitfalls and see if there are more